### PR TITLE
Thermometer: LM35/LM335/TMP36 max raw value 1023

### PR DESCRIPTION
**Pull request by fivdi**
Fixes https://github.com/rwaldron/johnny-five/issues/1298

### CODE REVIEW

**Comment by fivdi** [comment]
The test that's failing here is related to animations on OSX with Node.js v0.10:
```
Testing animation.js.......F..............................
>> Animation -- Servo - keyframeEasing
>> Error: 0 == true
>> at Object.exports.Animation -- Servo.keyframeEasing (test/animation.js:263:10)
>> at Object.exports.Animation -- Servo.setUp (test/animation.js:97:5)
```
The thermometer tests are fine:
```
Testing thermometer.js................................................................................................OK
```
Note that it wasn't necessary to modify the thermometer tests as they all passed after changing the max. raw value of an adc conversion from 1024 to 1023. This is a relatively small change in the max. raw value and only results in a relatively small change in the temperature reading. The tests use [math.round](https://github.com/rwaldron/johnny-five/blob/v0.10.4/test/thermometer.js#L46-L81) which means small changes in temperature are not relevant for the tests.

File: lib/thermometer.js [hunk 1]
```diff
@@ -343,7 +343,7 @@ var Controllers = {
     toCelsius: {
       value: function(raw) {
         // Analog Reference Voltage
-        var mV = this.aref * 1000 * raw / 1024;
```
Comment (rwaldron) [inline lib/thermometer.js]: lol. WHOOPS!

**Comment by rwaldron** [comment]
How do you feel about writing some additional tests to cover this against future regression? wink-wink, nudge-nudge

**Comment by fivdi** [comment]
Ok, I'll take a look.

**Comment by fivdi** [comment]
> How do you feel about writing some additional tests to cover this against future regression?

Regression tests added with 0ffe5fc8fd502e060704248e44df4a8ce036f74f

**Comment by fivdi** [comment]
The AppVeyor build times out after 60 minutes but I have no idea why. The timeout doesn't appear to be related to this PR.

**Comment by rwaldron** [comment]
Looks like bogus fails? It's passing fine now https://ci.appveyor.com/project/rwaldron/johnny-five/build/756

**Comment by rwaldron** [comment]
👏

**Pull request merged**